File: Framework-Samples/NHibernate/VisualBasic/Sample.Data/ManagerObjects/SupplierManager.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

Imports NHibernate
Imports Sample.Data.Generated.BusinessObjects
Imports Sample.Data.Generated.Base

Namespace Sample.Data.Generated.ManagerObjects
	'Public Partial Interface ISupplierManager
	'	Inherits IManagerBase(Of Supplier, System.Int32)
	'End Interface

	Partial Class SupplierManager
		Inherits ManagerBase(Of Supplier, System.Int32)
		Implements ISupplierManager
	End Class
End Namespace

File: Framework-Samples/NHibernate/CSharp/Sample.Data/ManagerObjects/SupplierManager.cs
using System;
using System.Collections.Generic;
using System.Text;

using NHibernate;
using Sample.Data.Generated.BusinessObjects;
using Sample.Data.Generated.Base;

namespace Sample.Data.Generated.ManagerObjects
{
    public partial interface ISupplierManager : IManagerBase<Supplier, int>
    {
	}
	
	partial class SupplierManager : ManagerBase<Supplier, int>, ISupplierManager
    {
	}
}